### PR TITLE
Slack

### PR DESCRIPTION
**Pull request by danhsiung**
Example Workflow: https://pipedream.com/@dan/slack-to-sms-p_OKClbN/edit

### CODE REVIEW

File: components/slack/new-message-in-conversation.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    slack,
+    conversations: {
+      type: "string[]",
+      label: "Conversations",
+      description: "Conversations you'd like to receive notifications for.",
+      optional: true,
+      async options({ prevContext }) {
+        let { types, cursor, userNames } = prevContext
+        if (types == null) {
+          scopes = await this.slack.scopes()
```
Comment (tjk) [inline components/slack/new-message-in-conversation.js:16]: declare scopes rather than using global?

File: components/slack/slack.app.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    },
+    sdk() {
+      return new WebClient(this.$auth.oauth_access_token)
+    },
+    async scopes() {
+      const resp = await this.sdk().auth.test()
+      if (resp.ok) {
+        return resp.response_metadata.scopes
+      } else {
+        console.log("Error getting scopes", resp.error)
```
Comment (tjk) [inline components/slack/slack.app.js:18]: why not throw here?

Comment (tjk) [reply components/slack/slack.app.js:18]: yea generally either errors should be handled better (which they aren't) or just throw so errors properly propagate

File: components/slack/new-message-in-conversation.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      // Due to inconsistencies with the shape of message_changed and message_deleted
+      // events, we are ignoring them for now. If you want to handle these types of
+      // events, feel free to change this code!!
+      console.log("Ignoring message with subtype.")
+      return
+    }
+    if (this.ignoreMyself && event.user == this.slack.mySlackId()) {
+      return
+    }
+    if (this.resolveNames) {
```
Comment (tjk) [inline components/slack/new-message-in-conversation.js:80]: would it make sense to use `db` to cache resolved names? it's bit more code but possibly much, much less wasteful

Comment (tjk) [reply components/slack/new-message-in-conversation.js:80]: although if cache could be out of date, might be a class of issues we'd rather avoid... for high volume slacks though would rather at least limit how often we end up resolving the same thing

File: components/slack/new-message-in-conversation.js (hunk 1 above)
Comment (danhsiung) [reply components/slack/new-message-in-conversation.js]: fixed

File: components/slack/slack.app.js (hunk 2 above)
Comment (danhsiung) [reply components/slack/slack.app.js:18]: fixed by throwing resp.error

File: components/slack/new-message-in-conversation.js (hunk 3 above)
Comment (danhsiung) [reply components/slack/new-message-in-conversation.js:80]: do you see this as a blocker? what would you propose? refresh once a day?

**Pull request merged**